flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/sink/FlinkKafkaSinkContext.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.kafka.sink;
+
+import org.apache.flink.annotation.Internal;
+
+/** Context providing information of the kafka record target location. */
+@Internal
+public interface FlinkKafkaSinkContext {
```

Comment:
I am open to renaming it but moving it to the `KafkaProducerRecordSerializationSchema` feels a bit strange because I would expect everything in `KafkaProducerRecordSerializationSchema` needs to be implemented by the user and the context will always come from us.

Reply:
That argument doesn't hold for `KafkaSerializationSchema$InitializationContext`, `SinkWriter$Context`, or `Sink$InitContext/ProcessingTimeService`. I'm open for alternative ways; I'm mostly aiming for consistency.